src/Components/IssuesControl.jsx

```diff
@@ -0,0 +1,239 @@
+import React, {useEffect, useState} from 'react'
+import {useLocation} from 'react-router'
+import {Link} from 'react-router-dom'
+import Paper from '@mui/material/Paper'
+import Typography from '@mui/material/Typography'
+import {makeStyles} from '@mui/styles'
+import {ControlButton} from './Buttons'
+import debug from '../utils/debug'
+import {
+  addHashParams,
+  getHashParams,
+  removeHashParams,
+} from '../utils/location'
+import {getIssue, getComment} from '../utils/GitHub'
+import CommentIcon from '../assets/2D_Icons/Comment.svg'
+import NavPrevIcon from '../assets/2D_Icons/NavPrev.svg'
+import NavNextIcon from '../assets/2D_Icons/NavNext.svg'
+
+
+/**
+ * The IssuesControl is a button that toggles display of issues for
+ * the currently selected element.  On load, this component also reads
+ * the current URL hash and fetches a related issue from GitHub, as
+ * well as adds a hash listener to do the same whenever the hash
+ * changes.
+ *
+ * @param {Object} viewer The viewer object from IFCjs.
+ * @return {Object} React component
+ */
+export default function IssuesControl({viewer}) {
+  const location = useLocation()
+  const [isDialogDisplayed, setIsDialogDisplayed] =
+        useState(parseHashParams(location) ? true : false)
+  const [text, setText] = useState('')
+  const prev = useState(window.location.href)
+  const [next, setNext] = useState(null)
+  useEffect(() => {
+    if (location) {
+      const p = parseHashParams(location)
+      if (p && p.issueId) {
+        if (Number.isInteger(p.commentId)) {
+          (async () => {
+            await showComment(p.issueId, p.commentId, setText, setNext)
+          })()
+        } else {
+          (async () => {
+            await showIssue(p.issueId, setText, setNext)
+          })()
+        }
+      }
+    }
+  }, [location])
+
+
+  // TODO: do a fetch to validate content before displaying panel
+  const showIssues = (doShow) => {
+    if (doShow) {
+      addHashParams(window.location, ISSUE_PREFIX, {id: 8})
+      setIsDialogDisplayed(true)
+    } else {
+      removeHashParams(window.location, ISSUE_PREFIX)
```

Comment:
at the moment the app crashes when the issue is closed.
it removes the hash parameter but seems to have trouble when the text is set to null.
<img width="331" alt="image" src="https://user-images.githubusercontent.com/3433606/156636210-8398b7d1-73ba-4df6-9d2b-3fe8ae3cd0de.png">